### DISCUSSION TITLE
1.24.0

### DISCUSSION THREAD

**Discussion by dani-garcia** [dani-garcia/vaultwarden#2275] [General]
⚠️  **Reminder**: If you are still using the `bitwardenrs/server*` Docker images, you need to migrate to the new `vaultwarden` image. Check https://github.com/dani-garcia/vaultwarden/discussions/1642 for an explanation. The old images will not receive any new updates any longer.

## What's Changed
* Add support for external icon services by @jjlin in https://github.com/dani-garcia/vaultwarden/pull/2158
  * Add config option to set the HTTP redirect code for external icons by @jjlin in https://github.com/dani-garcia/vaultwarden/pull/2188
  * Add support for legacy HTTP 301/302 redirects for external icons by @jjlin in https://github.com/dani-garcia/vaultwarden/pull/2218
* Support all DB's for Alpine and Debian by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/2172
* Add support for API keys by @jjlin in https://github.com/dani-garcia/vaultwarden/pull/2245
* Basic ratelimit for user login (including 2FA) and admin login by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/2165
* Upgrade Feature-Policy to Permissions-Policy by @iamdoubz in https://github.com/dani-garcia/vaultwarden/pull/2228
* Set `Expires` header when caching responses by @RealOrangeOne in https://github.com/dani-garcia/vaultwarden/pull/2182
* Increase length limit for email token generation by @jjlin in https://github.com/dani-garcia/vaultwarden/pull/2257
* Small changes to icon log messages. by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/2170
* Bump rust version to mitigate CVE-2022-21658 by @dscottboggs in https://github.com/dani-garcia/vaultwarden/pull/2255
* Fixed #2151 by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/2169
* Fixed issue #2154 by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/2194
* Fix issue with Bitwarden CLI. by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/2197
* Fix emergency access invites for new users by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/2217
* Sync global_domains.json by @jjlin in https://github.com/dani-garcia/vaultwarden/pull/2156
* Sync global_domains.json by @jjlin in https://github.com/dani-garcia/vaultwarden/pull/2171

## New Contributors
* @iamdoubz made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/2228
* @dscottboggs made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/2255

**Full Changelog**: https://github.com/dani-garcia/vaultwarden/compare/1.23.1...1.24.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dani-garcia/vaultwarden/releases/tag/1.24.0'>1.24.0</a>.</em>

**Comment by Tom41234** [comment]
Thanks for your amazing work!
